examples/parser/Parser/Http.roc

```diff
@@ -0,0 +1,257 @@
+interface Parser.Http
+    exposes [
+        Request,
+        Response,
+        request,
+        response,
+    ]
+    imports [
+        Parser.Core.{ Parser, ParseResult, map, apply, skip, const, oneOrMore, many },
+        Parser.Str.{
+            RawStr,
+            oneOf,
+            string,
+            codeunit,
+            parseStr,
+            codeunitSatisfies,
+            strFromRaw,
+            anyRawString,
+        },
+    ]
+
+# https://www.ietf.org/rfc/rfc2616.txt
+Method : [Options, Get, Post, Put, Delete, Head, Trace, Connect, Patch]
+
+HttpVersion : Str
+
+Request : {
+    method : Method,
+    uri : Str,
+    httpVersion : HttpVersion,
+    headers : List [Header Str Str],
+    body : List U8,
+}
+
+Response : {
+    httpVersion : HttpVersion,
+    statusCode : Str,
+    status : Str,
+    headers : List [Header Str Str],
+    body : List U8,
+}
+
+method : Parser RawStr Method
+method =
+    oneOf [
+        string "OPTIONS" |> map \_ -> Options,
+        string "GET" |> map \_ -> Get,
+        string "POST" |> map \_ -> Post,
+        string "PUT" |> map \_ -> Put,
+        string "DELETE" |> map \_ -> Delete,
+        string "HEAD" |> map \_ -> Head,
+        string "TRACE" |> map \_ -> Trace,
+        string "CONNECT" |> map \_ -> Connect,
+        string "PATCH" |> map \_ -> Patch,
+    ]
+
+expect parseStr method "GET" == Ok Get
+expect parseStr method "DELETE" == Ok Delete
+
+# TODO: do we want more structure in the URI, or is Str actually what programs want anyway?
+# This is not a full URL!
+#        Request-URI    = "*" | absoluteURI | abs_path | authority
+RequestUri : Str
+
+requestUri : Parser RawStr RequestUri
+requestUri =
+    codeunitSatisfies \c -> c != ' '
+    |> oneOrMore
+    |> map strFromRaw
+
+sp = codeunit ' '
+crlf = string "\r\n"
+
+# TODO: The 'digit' and 'digits' from Parser.Str are causing repl.expect to blow up
```

Comment:
do you have an example of this?

Reply:
Don't have one prepared but I can try to reproduce